plugins/Login/Security/BruteForceDetection.php

```diff
@@ -68,6 +71,10 @@ public function isEnabled()
 
     public function addFailedAttempt($ipAddress, $login = null)
     {
+        if (empty($login) && Request::isRootRequestApiRequest()) {
+            $login = self::API_LOGIN_PLACEHOLDER;
```

Comment:
would this cause any issue when there is an API call and it maybe searches for `isUserLoginBlocked(null)`? Could there be other side effects? 

Maybe generally when login is null it's likely API anyway?

Reply:
> would this cause any issue when there is an API call and it maybe searches for isUserLoginBlocked(null)? Could there be other side effects?

I don't think it's expected to search for `isUserLoginBlocked(null)`. The query that searches for logins uses `login = ?` not `login is NULL` so I wouldn't expect that work anyway.

> Maybe generally when login is null it's likely API anyway?

I think it could also be a tracker request?

Reply:
Possible re tracking request 👍 . Generally, I think it be better to add a column `source` or something similar just to prevent random side effects or bugs. If I see this right indeed `isUserLoginBlocked` is only called when there is a login so far. It just could also cause maybe side effects in the future as it may not be expected behaviour to have a login set where there is none and generally it's maybe not too important of an issue to introduce possible side effects in the future?

Reply:
A `source` column would probably be better. We could put the api method/module there as well. It wouldn't take too long to add, I'll look into it tomorrow.

Reply:
btw @diosmosis as the issue is not too high priority we could also just close the PR and work on it another day

Reply:
It won't take very long, I'll get to it when I have some time.